### ISSUE TITLE
Duplicate declaration $properties_file project.pp vs resource_source.pp

### ISSUE THREAD

**Issue by grafjo** [puppet-community/puppet-rundeck#83]
Using this manifest:
```puppet
rundeck::config::project { 'foo':
  resource_sources => {
    'foo' => {
      project_name             => 'foo',
      include_server_node => false,
      resource_format        => 'resourceyaml',
      source_type               => 'url',
      url                               => 'https://myservice',
      url_timeout                 => '30',
      url_cache                   => true,
    }
  }
}
```
triggers following error:
```
==> default: Error: Duplicate declaration: File[/var/rundeck/projects/foo/etc/project.properties] is already declared in file /tmp/vagrant-puppet-3/modules-0/rundeck/manifests/config/project.pp:78; cannot redeclare at /tmp/vagrant-puppet-3/modules-0/rundeck/manifests/config/resource_source.pp:199 on node localhost
==> default: Error: Duplicate declaration: File[/var/rundeck/projects/foo/etc/project.properties] is already declared in file /tmp/vagrant-puppet-3/modules-0/rundeck/manifests/config/project.pp:78; cannot redeclare at /tmp/vagrant-puppet-3/modules-0/rundeck/manifests/config/resource_source.pp:199 on node localhost
```

Who is responsible for creating the directory structure and the $properties_file?
Is it a good idea to use a resource_source.pp without a project.pp? I think resource_source.pp can not life without an project.pp because of the required param project_name by resource_source.
IMHO the project.pp should be responsible for $properties_file and the folder structure. So we can remove some duplicated code from resource_source.pp and make resource_source.pp a private define.

What do you think?

**Comment by dalisch** [comment]
I'm currently dealing with this error as well.

My thoughts are that although a typical use case (ourselves included) would be to have a resource model shared across projects, that isn't the way Rundeck is currently designed, so the resource model should be handled by the project.  There is a feature request for having job libraries that span multiple projects as well (https://trello.com/c/jtFcbflm), but currently jobs seem to be constrained to the project.

**Comment by liamjbennett** [comment]
The duplication error should have been fixed by this: d3ee42f

But I will keep this open because I think I agree that there needs to be some further refactoring done here.
